examples/cis3-nft-sponsored-txs/src/lib.rs

```diff
@@ -1304,6 +1355,29 @@ fn contract_set_implementor<S: HasStateApi>(
     Ok(())
 }
 
+// Helper function to decode a hex string into bytes.
+#[allow(dead_code)]
+fn decode_hex(s: &str) -> Result<Vec<u8>, CustomContractError> {
+    if s.len() % 2 != 0 {
+        Err(CustomContractError::ParseError)
+    } else {
+        (0..s.len())
+            .step_by(2)
+            .map(|i| u8::from_str_radix(&s[i..i + 2], 16).map_err(|e| e.into()))
+            .collect()
+    }
+}
+
+// Helper function to encode bytes into a hex string.
+#[allow(dead_code)]
+fn encode_hex(bytes: &[u8]) -> String {
+    let mut s = String::with_capacity(bytes.len() * 2);
+    for &b in bytes {
+        s = format!("{}{:02x}", s, b);
+    }
+    s
+}
```

Comment:
And this should be the display implementation, e.g., 
```
impl std::fmt::Display for HashSha2256 {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        for b in self.0 {
            write!(f, "{:02x}", b)?;
        }
        Ok(())
    }
}
```

in concordium-std directly (or contracts-common if the types are defined there).

Then you can just do `to_string`